File: Zarina/TaZQL/src/ast/type/UndefinedType.java
package ast.type;

public class UndefinedType extends Type {
	
	public UndefinedType() { }
	
	@Override
	public String toString() {
		return "undefined"; 
	}
	
	@Override
	public <T> T accept(ITypeVisitor<T> visitor) {
		return visitor.visit(this);
	}
		
	@Override
	public boolean isCompatibleToType(Type type) {
		return false;
	}
}

File: FelipezConde/QuestionnaireLanguage/AST/Types/UndefinedType.cs

namespace AST.Types
{
    public class UndefinedType : Type
    {
        public override bool IsEqual(Types.Type type)
        {
            return false;
        }

        public override bool IsUndefined()
        {
            return false;
        }

        public override T Accept<T>(Visitors.ITypeVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public override string ToString()
        {
            return "undefined";
        }
    }
}
